integration/embedded_test.go

```diff
@@ -76,3 +79,60 @@ func TestEmbedded(t *testing.T) {
 	cancel()
 	<-done
 }
+
+func TestEmbeddedTLS(t *testing.T) {
```

Comment:
I wonder if we could reduce a number of duplications with TestEmbedded by using subtests